### PR TITLE
Alistair's creation

### PR DESCRIPTION
**Pull request by Alistair1080**

### CODE REVIEW

File: src/scrabble.js [hunk 1]
```diff
@@ -1,5 +1,129 @@
...
+      }
+    } else {
+      score = 0
+      break
+    }
+  }
+  return score
+}
+
+// The functions below are only called when checking an inputted word has valid bracket use
```
Comment (EwbiDev) [inline src/scrabble.js:88]: I like that you have a lot of functions! Easy to get a general idea what your code is doing quickly!

File: src/scrabble.js [hunk 2]
```diff
@@ -1,5 +1,129 @@
...
+}
+
+function bChecker(word, x) {
+  let check = false
+  for (let i = x; i >= 0; i--) {
+    if (word[i] === '{') {
+      check = true
+    }
+  }
+  return check
```
Comment (EwbiDev) [inline src/scrabble.js:126]: Didn't think about checking through the word the word backwards to do stuff, nice idea!

Comment (EwbiDev) [reply src/scrabble.js:126]: Question on the backwards loops, do they only work going backwards? These functions don't strike me as needing to do so in these particular cases.

File: src/scrabble.js [hunk 3]
```diff
@@ -1,5 +1,129 @@
...
+    return 0
+  }
+}
+
+// Responsable for working out the total score
+function calculator(word) {
+  let score = 0
+  let multiplier = 1
+  // Indexs through each letter
+  for (let x = 0; x < word.length; x++) {
```
Comment (NazarTymiv) [inline src/scrabble.js:47]: Could be better to use split method for word variable and take array of each letter in the word and after that use forEach

File: src/scrabble.js [hunk 4]
```diff
@@ -1,5 +1,129 @@
...
+        score = 0
+        break
+      }
+    } else {
+      score = 0
+      break
+    }
+  }
+  return score
+}
```
Comment (EwbiDev) [inline src/scrabble.js:86]: This function does get a bit long. Using `switch` may be a good use case here as an alternative structure:
```js
switch(letter) {
case '[':
  // logic goes here
case '{':
  // logic goes here
...
}

File: src/scrabble.js [hunk 5]
```diff
@@ -1,5 +1,129 @@
...
+  let multiplier = 1
+  // Indexs through each letter
+  for (let x = 0; x < word.length; x++) {
+    const letter = word.charAt(x).toUpperCase()
+
+    if (scores[letter] !== undefined) {
+      score += scores[letter] * multiplier
+    } else if (letter === '[') {
+      if (squareBracketChecker(word, x) === true) {
+        multiplier = multiplier * 3
```
Comment (NazarTymiv) [inline src/scrabble.js:54]: Can be multiplier *= 3

File: src/scrabble.js [hunk 6]
```diff
@@ -1,5 +1,129 @@
...
+  let check = false
+  for (let i = x; i >= 0; i--) {
+    if (word[i] === '{') {
+      check = true
+    }
+  }
+  return check
 }
 
 module.exports = scrabble
```
Comment (EwbiDev) [inline src/scrabble.js:129]: Well done on completed the extended exercise! 😄 Also oops I didn't mean to highlight the whole code here.

File: src/scrabble.js [hunk 7]
```diff
@@ -1,5 +1,129 @@
...
+// console.log(scrabble("[hi]"))
+
+// Main function - checks the input is a string then sends the parameter value to the calculator function.
+function scrabble(word) {
+  if (typeof word === 'string') {
+    return calculator(word)
+  } else {
+    return 0
+  }
+}
```
Comment (NazarTymiv) [inline src/scrabble.js:40]: Good way for doing this check and call another function :)

**Review by NazarTymiv — COMMENTED** [review]
Good job, nice work :)

**Review by EwbiDev — COMMENTED** [review]
Good job!

File: src/scrabble.js [hunk 8]
```diff
@@ -1,5 +1,79 @@
...
+  Z: 10
+}
+
+// Main function - checks the input is a string then sends the parameter value to the calculator function.
+function scrabble(word) {
+  if (typeof word === 'string') {
+    return calculator(word)
+  } else {
+    return 0
+  }
```
Comment (jdm1991) [inline src/scrabble.js:39]: Really easy to read dude. Awesome work.

File: src/scrabble.js [hunk 9]
```diff
@@ -1,5 +1,79 @@
...
+  // indexs through each letter
+  for (let x = 0; x < word.length; x++) {
+    const letter = word.charAt(x).toUpperCase()
+    // console.log(letter)
+    // console.log(alphabet[letter])
+    if (scores[letter] !== undefined) {
+      score += scores[letter] * multiplier
+    } else {
+      score = 0
+      break
```
Comment (jdm1991) [inline src/scrabble.js:82]: Curious to understand your structure on this. Looks really clean and defined.

**Pull request closed without merging**